src/richie/apps/search/elasticsearch.py

```diff
@@ -0,0 +1,96 @@
+"""
+Build an interoperability layer that allows Richie to work with ElasticSearch 6
+just like it does with ElasticSearch 7.
+"""
+# pragma pylint: disable=W0102,W0221
+from elasticsearch import Elasticsearch, Transport
+from elasticsearch.client import IndicesClient
+from elasticsearch.helpers import bulk
+
+DOC_TYPE = "_doc"
+
+
+class ElasticsearchClientCompat7to6(Elasticsearch):
+    """
+    Compatibility wrapper around the Elasticsearch client from elasticsearch-py that
+    handles incompatibilities to let Richie run ES6 and ES7.
+    """
+
+    def __init__(self, hosts=None, transport_class=Transport, **kwargs):
+        """
+        Instantiate the actual Elasticsearch client, then use it to detect the version
+        of Elasticsearch we're working with.
+        """
+        super().__init__(hosts=hosts, transport_class=transport_class, **kwargs)
+        # Store the ES version information in memory so we don't have to re-make the
+        # info request constantly.
+        self.__es_version__ = self.info()["version"]["number"][:1]
+
+    def search(self, body=None, index=None, params={}, **kwargs):
+        """
+        Patch the "value" & "relation" dict in place of the int returned by ES6 for
+        search query hits total count.
+        """
+
+        search_response = super().search(
+            body=body, index=index, params=params, **kwargs
+        )
+
+        if self.__es_version__ == "6":
+            search_response["hits"]["total"] = {
+                "value": search_response["hits"]["total"],
+                "relation": "eq",
+            }
+
+        return search_response
+
+
+class ElasticsearchIndicesClientCompat7to6(IndicesClient):
+    """
+    Compatibility wrapper around the ES IndicesClient from elasticsearch-py that
+    handles incompatibilities to let Richie run ES6 and ES7.
+    """
+
+    def get_mapping(self, index=None, params={}):
```

Comment:
declaring {} as default param is a Python anti-pattern. Not sure where it comes from and why you need it? It is also causing E1120 if I understand correctly so maybe there is a better way?

Reply:
I tried another way that does not cause E1120. The decorator used by the elasticsearch-py lib on their methods makes the class painful to subclass, hence these shenanigans...

Reply:
Your modification solves the anti-pattern. So can you remove E1120 comments now or it still raises issues with pylint?

Reply:
I did remove it. The `W0221` is a real need as it expected that my methods and the ones from the lib do not have the same number of arguments (as they do not have the doc type argument)